### PR TITLE
Separate Player and Enemy logic; add Health and Projectile

### PR DESCRIPTION
**Pull request by logar16**
Trying to keep player and enemies from being too tied together.
Also added a health and projectile component for combat later.

### CODE REVIEW

File: Assets/Scripts/Health.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace FunMath
+{
+    public class Health : MonoBehaviour
+    {
+        [SerializeField]
+        [Range(0, 100)]
+        int health = 10;
+
+        public void TakeDamage(int damage)
```
Comment (ShumWengSang) [inline Assets/Scripts/Health.cs]: Probably rename it to modify health instead, to shelter against postive/negative number bugs

File: Assets/Scripts/Projectile.cs
Comment (ShumWengSang) [inline Assets/Scripts/Projectile.cs:1]: Is movement coming later?

**Review by ShumWengSang — APPROVED** [review]

File: Assets/Scripts/PlayerCharacter.cs [hunk 2]
```diff
@@ -2,14 +2,21 @@
...
 namespace FunMath
 {
-    public class PlayerCharacter : BaseCharacter
+    public class PlayerCharacter : MonoBehaviour
     {
         [SerializeField]
         [Range(0.1f, 2f)]
         float speed = 1f;
 
+        protected Rigidbody2D rb;
```
Comment (LuffaKing) [inline Assets/Scripts/PlayerCharacter.cs:11]: We should use a better name here. Maybe full name to avoid confusion.

File: Assets/Scripts/Projectile.cs
Comment (logar16) [reply Assets/Scripts/Projectile.cs:1]: Yes

File: Assets/Scripts/PlayerCharacter.cs (hunk 2 above)
Comment (logar16) [reply Assets/Scripts/PlayerCharacter.cs:11]: Will rename in the next PR when I do a major rehaul.

**Pull request merged**